### ISSUE TITLE
rpg-skilltrees: meet with action management, integrate skilltrees into a view action

### ISSUE THREAD

**Issue by KabirDubey** [uchicago-cs/chiventure#1510]
The action management module works with the conditions and effects involved to manipulate a player in chiventure. They designed a fourth "kind" of action (called VIEW) that presumably makes a player view items (further documentation pending).  This utilizes a `do_self_action` function that executes VIEW actions when given various player attributes. From lines 157-174 of the header file:

```
/* A function that executes KIND 4 actions (ACTION <player>)
 *
 * Parameters:
 * - c: A context struct encapsulating the shared state in chiventure
 * - a: An action type struct
 * - p: A player struct
 * - obj: An enum describing what self-related object (STATS, etc)
 *        needs to be acted on
 * - ret_string: A pointer to a string describing the result of the function
 *   - NOTE: THIS STRING IS MALLOCED AND MUST BE FREED BY USERS OF THIS FUNCTION
 *
 * Returns:
 * - 0 upon success, success string as an out parameter
 * - WRONG_KIND if the action type has the wrong kind, failure string as an out parameter
 * - 6 if conditions for the action haven't been met, failure string as an out parameter
 */
int do_self_action(chiventure_ctx_t *c, action_type_t *a, player_t *p,
                  self_action_object obj, char **ret_string);
```

We will meet with the action management team to discuss what is needed from the skilltrees team to make this possible for skilltrees. We will create functions in the skilltrees module that return skilltree information appropriately for action mangement.

**Comment by KabirDubey** [comment]
Meeting outcomes:
1. Make a `display_skills()` function in the `skilltree` file that iterates through the skilltree and returns a string containing the prerequisite level and any other appropriate info about skills. 
2. Send that to action management to integrate. Action management will handle testing.

**Comment by anchristensen** [comment]
After further conversation with action management we have decided to add two new functions to support their work: 1) the display skilltree function that returns high level info about skills within the skilltree, as @KabirDubey described and 2) a function that takes in a skill and prints its description

**Comment by anchristensen** [comment]
Brief update: both functions have been implemented and some tests have been added. Right now one of the tests is failing, once we figure that test out this issue should be done. We're hoping to get the pull request in within the next day or so.

**Comment by anchristensen** [comment]
Closing statement: Pull request #1575 will finalize this issue. The action management team described to us a module of theirs which works with the conditions and effects involved to manipulate a player in chiventure. They designed a fourth kind of action (called VIEW) that makes a player view items/info. They needed two functions from us to effectively be able to display skilltrees: The first, given a skilltree, returns a char * with basic information about that skilltree, for instance each node's skill name, the current skill level, and the necessary prereq level. The second takes in a skill name, the skilltree containing the skill, and prints its description (e.g. "Throws a fireball"). These two functions provide high-level information for the described module. This PR includes the code for both functions, as well as tests for both of them. I wrote both functions and tests.
